client/src/pages/Programmes/ProgrammesPage.js

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect, useState } from "react";
+import { Flex, Text, SimpleGrid, IconButton, usePrevious } from "@chakra-ui/core";
+import { Link, useParams } from "react-router-dom";
+import { AiFillHome } from "react-icons/ai";
+import { SearchBar, ProgrammeCard } from "../../components";
+import MomentumClient from "../../common/MomentumClient";
+import ExistingProgramme from "./ExistingProgramme";
+import EmptyProgramme from "./EmptyProgramme";
+import NewProgramme from "./NewProgramme";
+import useProgrammmes from "./useProgrammes";
+
+const ProgrammesPage = () => {
+    const location = useParams();
+    const { programmeDegrees, fetchAllProgrammes } = useProgrammmes();
+    const [numOfProgrammes, setNumOfProgrammes] = useState(programmeDegrees.length);
+
+    useEffect(fetchAllProgrammes, [numOfProgrammes]);
```

Comment:
Not sure about this but isn't `numOfProgrammes` redundant because it directly depends on `programDegrees` and when you add a new course you call the API and then you call fetchProgramme which would update this?

Reply:
I tried this but it still wasn't updating D: maybe we can refactor afterwards?

Reply:
Yea probably better to refactor afterwards, and just get the core functionality done asap :shipit:

Reply:
Final words before it all went wrong 😆